### PR TITLE
Add Developer Guide

### PR DESCRIPTION
**Pull request by dhilipkumars**
Added Developer Guide with reference to issue https://github.com/coreos/etcd-operator/issues/456

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ Users need to create multiple operators in different namespaces to manage etcd c
 
 ![etcd Operator demo](https://raw.githubusercontent.com/coreos/etcd-operator/master/doc/gif/demo.gif)
 
+## Developer Guide
```
Comment (hongchaodeng) [inline README.md]: Don't make a section.

Let's simply add a sentence above "Requires" saying:
```
Read [developer guide](link) for ...
```

File: doc/devel/development.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Developer guide 
+This document explains how to setup your dev environment. 
+
+## Prerequisite 
+- kubernetes 1.4 
+- go-lang version 1.7
```
Comment (hongchaodeng) [inline doc/devel/development.md]: Just "Go 1.7"

File: doc/devel/development.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+- docker 
+- glide 
+
+## Setup
+### Step 1
```
Comment (hongchaodeng) [inline doc/devel/development.md]: Most of the setup git stuff is unrelated to etcd operator, and we shouldn't teach people how to use git here. Please remove them and leave only related stuff, e.g. how to glide install, build, etc.

File: hack/build/operator/local_build [hunk 4]
```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env bash
```
Comment (hongchaodeng) [inline hack/build/operator/local_build]: This file is 99% similar to "hack/test".
Why create this file?

File: doc/devel/development.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Developer guide
```
Comment (xiang90) [inline doc/devel/development.md]: a minor styling issue: put an empty line between headlines and its content. Thanks!

File: doc/devel/development.md (hunk 3 above)
Comment (hongchaodeng) [inline doc/devel/development.md]: On a second thought, I think we have documented how to build stuff in "hack/release". Maybe we can just add a link to it.

Other than build, we can also talk about code style, how to test in this doc. We also have a doc about e2e already.

**Comment by dhilipkumars** [comment]
Apologies for the delay, i had somehow managed to miss this PR. Let me know if you folks are still interested in this doc.

**Comment by hongchaodeng** [comment]
@dhilipkumars 
That's fine. We are definitely interested in the dev docs.

**Comment by hongchaodeng** [comment]
We want to prioritize dev guide. 
Any progress on this? Otherwise we will cherry pick this and change it to another PR tomorrow.

**Comment by dhilipkumars** [comment]
Did a re-base and brough it like 250+ commits to this PR, i will raise another one which will be easier for review.  Closing this one

**Pull request closed without merging**